src/tests/core/engine/app_engine_processor_test.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+from unittest.mock import patch
+
+from src.core.engine.app_engine_processor import AppEngineProcessor
+from src.core.engine.processors.kitchen_simulator import KitchenSimulator
+
+
+class TestAppEngineProcessor(unittest.TestCase):
+    @patch("src.core.engine.app_engine_processor.AppProcessorConfig")
+    @patch("src.core.engine.app_engine_processor.KitchenSimulator")
+    def test_mock_in_app_context(self, mock_kitchen, mock_config):
+
+        app_engine_processor = AppEngineProcessor()
+        self.assertEqual(*app_engine_processor.app_context.processors, mock_kitchen())
+        mock_config.assert_called_once()
+
+    @patch(
+        "src.core.engine.app_engine_processor.KitchenSimulator.start",
+        wraps=KitchenSimulator,
+    )
+    def test_start_in_context_run_once(self, mock_kitchen):
```

Comment:
This test is not ok, you should mock `KitchenSimulator` and make it to return your own instance using a `build_kitchen_simulator_processor` function and then check if the start from that mock was called when you ran `app_engine_processor.start()`